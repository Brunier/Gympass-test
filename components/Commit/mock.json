{
  "sha": "942a765b53662936679b8119ed4533d972298d2a",
  "commit": {
    "author": {
      "name": "Pedro Bullo Junior",
      "email": "pedro.bj@hotmail.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "committer": {
      "name": "GitHub",
      "email": "noreply@github.com",
      "date": "2019-01-21T22:59:19Z"
    },
    "message": "Workaround empty child",
    "tree": {
      "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
      "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
    },
    "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
    "comment_count": 0,
    "verification": {
      "verified": true,
      "reason": "valid"
    }
  },
  "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
}
