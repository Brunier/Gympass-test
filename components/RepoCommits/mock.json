{
  "data" : [
    {
      "sha": "942a765b53662936679b8119ed4533d972298d2a",
      "commit": {
        "author": {
          "name": "Pedro Bullo Junior",
          "email": "pedro.bj@hotmail.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2019-01-21T22:59:19Z"
        },
        "message": "Workaround empty child",
        "tree": {
          "sha": "81c34b008db2a4fda4df20404a498e0127e4a817",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/81c34b008db2a4fda4df20404a498e0127e4a817"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/942a765b53662936679b8119ed4533d972298d2a",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/942a765b53662936679b8119ed4533d972298d2a"
    },
    {
      "sha": "c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
      "commit": {
        "author": {
          "name": "matc4",
          "email": "matias.s.cortes@gmail.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "committer": {
          "name": "GitHub",
          "email": "noreply@github.com",
          "date": "2018-09-13T12:34:04Z"
        },
        "message": "Merge pull request #100 from nesc58/masteFix: Text elements / transform + clipPath attribute added",
        "tree": {
          "sha": "97dae56ead52ab59e12b98848f4e38d89265040f",
          "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/trees/97dae56ead52ab59e12b98848f4e38d89265040f"
        },
        "url": "https://api.github.com/repos/Gympass/react-native-svg-uri/git/commits/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c",
        "comment_count": 0,
        "verification": {
          "verified": true,
          "reason": "valid"
        }
      },
      "url": "https://github.com/Gympass/react-native-svg-uri/commit/c6841a6e7d1a6ee5cf5b515d4faea6002e166f3c"
    }
  ]
}
