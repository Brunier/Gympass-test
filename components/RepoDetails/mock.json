{
  "name": "interview-test",
  "description": "Interview test project",
  "url": "https://api.github.com/repos/Gympass/interview-test",
  "commits_url": "https://api.github.com/repos/Gympass/interview-test/commits",
  "created_at": "2018-05-22T19:57:37Z",
  "updated_at": "2019-01-24T02:51:23Z",
  "language": "Javascript",
  "stars": 3,
  "watchers": 3,
  "forks": 4
}
