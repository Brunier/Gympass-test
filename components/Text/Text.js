import styled from 'styled-components';
import React from 'react';



let Text = styled.span`
  color: #585656;
  font-size: ${props => props.fontSize ? props.fontSize + "px"  : "20px"};
  font-weight: ${props => props.bold ? "bold" : "none"};
`

export default Text;

