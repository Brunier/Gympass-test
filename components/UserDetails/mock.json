{
  "repos": [
    {
      "name": "mindbody-api",
      "description": "A Ruby interface to the MindBody API v0.5",
      "url": "https://api.github.com/repos/Gympass/mindbody-api",
      "commits_url": "https://api.github.com/repos/Gympass/mindbody-api/commits",
      "created_at": "2018-04-09T17:04:01Z",
      "updated_at": "2018-04-13T21:52:29Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "rails-intl-tel-input",
      "description": "Integrate with intl-tel-input to provide international telephone numbers input and form helper.",
      "url": "https://api.github.com/repos/Gympass/rails-intl-tel-input",
      "commits_url": "https://api.github.com/repos/Gympass/rails-intl-tel-input/commits",
      "created_at": "2017-11-14T17:27:57Z",
      "updated_at": "2018-03-16T12:40:19Z",
      "language": "JavaScript",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    }
  ],
  "name": "gympass",
  "img": "https://avatars3.githubusercontent.com/u/26311162?v=4",
  "loading": false,
  "notFound": false
}
