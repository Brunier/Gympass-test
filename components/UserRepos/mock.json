{
  "data": [
    {
      "name": "mindbody-api",
      "description": "A Ruby interface to the MindBody API v0.5",
      "url": "https://api.github.com/repos/Gympass/mindbody-api",
      "commits_url": "https://api.github.com/repos/Gympass/mindbody-api/commits",
      "created_at": "2018-04-09T17:04:01Z",
      "updated_at": "2018-04-13T21:52:29Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "rails-intl-tel-input",
      "description": "Integrate with intl-tel-input to provide international telephone numbers input and form helper.",
      "url": "https://api.github.com/repos/Gympass/rails-intl-tel-input",
      "commits_url": "https://api.github.com/repos/Gympass/rails-intl-tel-input/commits",
      "created_at": "2017-11-14T17:27:57Z",
      "updated_at": "2018-03-16T12:40:19Z",
      "language": "JavaScript",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "devise",
      "description": "Flexible authentication solution for Rails with Warden.",
      "url": "https://api.github.com/repos/Gympass/devise",
      "commits_url": "https://api.github.com/repos/Gympass/devise/commits",
      "created_at": "2018-01-17T00:17:46Z",
      "updated_at": "2018-01-17T00:17:52Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    },
    {
      "name": "AvaTax-REST-V2-Ruby-SDK",
      "description": "Ruby SDK for AvaTax REST v2",
      "url": "https://api.github.com/repos/Gympass/AvaTax-REST-V2-Ruby-SDK",
      "commits_url": "https://api.github.com/repos/Gympass/AvaTax-REST-V2-Ruby-SDK/commits",
      "created_at": "2018-01-09T23:23:04Z",
      "updated_at": "2018-01-09T23:23:06Z",
      "language": "Ruby",
      "stars": 0,
      "watchers": 0,
      "forks": 0
    }
  ]
}
