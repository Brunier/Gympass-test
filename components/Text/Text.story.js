import React from 'react';
import { storiesOf } from '@storybook/react';
import Text from './Text';


storiesOf('Text', module)
    .add('Normal', () => (
        <Text>Texto Normal</Text>
    ))
    .add('FontSize x (10px)', () => (
        <Text fontSize={10}>Texto com 10px</Text>
    ))
    .add('Negrito', () => (
        <Text bold={true}>Texto em Negrito</Text>
    ))
