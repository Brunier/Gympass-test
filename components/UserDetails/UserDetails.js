import React from 'react';
import styled from "styled-components";
import Text from "../Text/Text";

const Container = styled.div`
  text-align: center;
  img {
    width: 7%;
    border-radius: 20px;
  }
  
 `;

export class UserDetails extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        return(
            <Container>
                <div>
                    <div>
                        <img src={this.props.user.img} />
                    </div>
                    <div>
                        <Text bold={true}>{this.props.user.name}</Text>
                        <Text bold={true}> - </Text>
                        <Text bold={true}>Mostrando {this.props.user.repos.length} reposit√≥rios no total</Text>
                    </div>

                </div>
            </Container>
        )
    }
}

export default UserDetails;
