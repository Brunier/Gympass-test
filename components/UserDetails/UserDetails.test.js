import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import mock from './mock'

// Components
import UserDetails from './UserDetails';
import Text from "../Text/Text";


function setup(props) {
    const wrapper = shallow(<UserDetails {...props} />);
    return { wrapper };
}


const user = mock;


describe('UserDetails Test Suite', () => {
    it('Imagem do usu치rio', () => {
        const { wrapper } = setup({user});
        expect(wrapper.find("img").prop("src")).toBe(user.img);
    });
    it('Nome do usu치rio', () => {
        const { wrapper } = setup({user});
        expect(wrapper.find(Text).at(0).text()).toBe(user.name);
    });
    it('Quantidade de reposit칩rios', () => {
        const { wrapper } = setup({user});
        expect(wrapper.find(Text).at(2).text().trim()).toBe(`Mostrando ${user.repos.length} reposit칩rios no total`);
    });

    it('Snapshot', () => {
        const tree = renderer.create(<UserDetails user={user}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
