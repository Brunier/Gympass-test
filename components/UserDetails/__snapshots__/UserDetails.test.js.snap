// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails Test Suite Snapshot 1`] = `
<div
  className="sc-bwzfXH cibgGO"
>
  <div>
    <div>
      <img
        src="https://avatars3.githubusercontent.com/u/26311162?v=4"
      />
    </div>
    <div>
      <span
        className="sc-bdVaJa fngNYB"
      >
        gympass
      </span>
      <span
        className="sc-bdVaJa fngNYB"
      >
         - 
      </span>
      <span
        className="sc-bdVaJa fngNYB"
      >
        Mostrando 
        2
         reposit√≥rios no total
      </span>
    </div>
  </div>
</div>
`;
